server/middlewares/substackDomainCache.ts

```diff
@@ -0,0 +1,54 @@
+import { type NextFunction, type Request, type Response } from 'express'
+import { getSld, getSubdomain } from '../../common/domain-utils.ts'
+import { buildClient } from '../src/client.ts'
+
+const substackDomainCache: Record<string, string> =
+  process.env.DEBUG
+    ? {
+        'www.polymorpher.localhost:3100': 'polymorpher.substack.com',
+        'localhost:3100': 'polymorpher.substack.com',
+        'localhost:3002': 'polymorpher.substack.com'
+      }
+    : {}
+
+const client = buildClient()
+
+const getDomain = (host: string): string => {
+  const res = host.split('://')
+  if (res.length === 0) {
+    return res[0]
+  } else {
+    return res[1]
+  }
+}
+
+const cache = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+  const host = req.get('host')
+
+  if (!host) {
+    return
+  }
+
+  let substackDomain = substackDomainCache[host]
```

Comment:
I don't think we need cache here. I will use a private RPC node for the server, hosted in the same zone on GCP, so the smart contract query time is negligible. If we really need to use cache, we should use LRU cache with timeout of 1 minute, since domain owners may change what they set on the contract.

Reply:
Removed cache